### PR TITLE
Change configuration format to be hierarchical and file based

### LINKED ISSUE #25: Task: remove requirement for creating a config.py file
Right now the project does not build by default since it has an import chain that ends in a file which does not exist in the repo.

Instead if we moved the config values to a json (or yaml or what-have-you) file, we could always build.

### Affected Functionality
This makes it difficult to run unit tests (and for first time setup).

### PR DESCRIPTION
**Pull request by Olivia5k**
Closes #25.
Closes #27.

### CODE REVIEW

File: default-config.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    id: "889952b3-7b44-148e-9323-f38ccabb2744"
+    secret: "2f55d379-a27c-1101-dbfe-25c70e44f330"
+  webhook:
+    secret: "6b3c2e7b-ce1f-776d-1ff2-aaf6322b5f6e"
+
+# Where the gamedata is located on your computer. This is used by the client to
+# figure out where to look for run files and autosaves.
+spire:
+  # TODO(olivia): Probably default to something Windows specific?
+  steamdir: ~/.steam/steam/steamapps/common/SlayTheSpire
```
Comment (Vgr255) [inline default-config.yml:105]: Windows default is `C:\Program Files (x86)\Steam\steamapps\common\SlayTheSpire`

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,3 @@
-config.py
 *.pyc
 data/
 last_run
```
Comment (Vgr255) [inline .gitignore:3]: Will need to ignore all `.yml` files that aren't `default-config.yml`

File: disc.py [hunk 3]
```diff
@@ -21,13 +21,13 @@ async def invoke(self, context: Context):
             return
         if self.flag:
             for guild in context.bot.guilds:
-                if guild.id == config.discord_servid:
+                if guild.id == config.discord.server_id:
                     break
             else:
                 return
-            mod_role = guild.get_role(config.moderator_role)
+            mod_role = guild.get_role(config.discord.moderator_role)
```
Comment (Vgr255) [inline disc.py:28]: This should always be an `int`

**Review by Vgr255 — CHANGES_REQUESTED** [review]

File: config.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    with open(filename, 'r') as stream:
+        # With _dynamic=False, any access on something we hadn't yet
+        # set will raise an exception. With _dynamic=True (the default) it
+        # would instead return None. This lets us know when we access things
+        # that aren't set.
+        conf = DotMap(yaml.safe_load(stream), _dynamic=False)
+        return conf
+
+# Load the default configuration file
+config = load_file("./default-config.yml")
```
Comment (zdylag) [inline config.py:34]: Python noob question: does this have types and properties? 

If not: could we have that?

Comment (thiderman) [reply config.py:34]: It follows whatever types we've set in the configuration file, but there's no validation or anything like that. I personally don't think that's super important for the configuration files, especially as long as the default file has the right things.

I also saw that you made #27 only after I had made this PR, haha.

File: default-config.yml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  json_indent: 2
+
+  # This email is visible in the footer of every page. For development contexts,
+  # this can be left empty.
+  business_email: <email not set>
+
+  # Websocket and webhook setup. TODO: Document.
+  websocket_client:
+    id: "889952b3-7b44-148e-9323-f38ccabb2744"
+    secret: "2f55d379-a27c-1101-dbfe-25c70e44f330"
```
Comment (brianyee) [inline default-config.yml:98]: Is it fine to have these in the repo?

Comment (Vgr255) [reply default-config.yml:98]: These are dummy UUIDs, they're invalid. Though it could be made more obvious

Comment (thiderman) [reply default-config.yml:98]: That's the TODO part, since I don't know that much about this part of the code.

Comment (Olivia5k) [reply default-config.yml]: Fixed in a new commit.

**Comment by Olivia5k** [comment]
@Vgr255 All outstanding comments and issues are resolved. Can we merge this now?

**Review by Vgr255 — APPROVED** [review]

**Comment by Vgr255** [comment]
This is now ready to merge. I will prepare the server to receive it sometime today, and will merge at that point.

**Pull request merged**